src/renderer/components/shared/service/ServiceForm.vue

```diff
@@ -168,7 +185,8 @@ export default {
         namespace: '',
         workloadType: null,
         workloadName: '',
-        forwards: []
+        forwards: [],
+        localAddress: null
```

Comment:
I would like to propose to store `localhost` as default value of `localAddress`^ to avoid construction as `this.service.localAddress || 'localhost'` and just use `this.service.localAddress`. Also we will be able to use a simple concatenation `${localAddress}:${forward.localPort}`

Reply:
That could work in theory, the only problem is that this field doesn't exist in the configuration currently, so we still can't assume that it exists unless we do some kind of "fixup" stage after loading a config to ensure it has this field in it.  Do you think that's the way to go?

Reply:
Another issue with defaulting to `localhost` is that we lose the ability to tell the difference between explicitly setting localhost and using the default.  It's possible in the future we might not use `localhost` as the default, perhaps we discover that `127.0.0.1`, or `0.0.0.0`, or `null` (let node.js choose) are better options.  If we've stuck this "default" value in there without some additional field indicating that it's not the default, then we'll be uncertain as to whether the user specifically wants `localhost` or they just left it at defaults.

Anyway, just some thoughts.  At the end of the day my primary concern is getting some version of this merged so I don't have run it from my fork any more, so I will do whatever it takes.

Reply:
Good points, I agree. I think it could be merges as is. 👍